[SMTChecker] Add option to choose contracts to be analyzed (#11261)
ethereum/solidity: docs/smtchecker.rst

There is no precise heuristic on how and when to split verification targets,
but it can be useful especially when dealing with large contracts.

Verified Contracts
==================

> By default all the contracts in the given sources are analyzed separately as

Reviewer: Maybe it is not needed, but I think we should state explicitly 
```suggestion
By default all the deployable contracts in the given sources are analyzed separately as
```

There is no precise heuristic on how and when to split verification targets,
but it can be useful especially when dealing with large contracts.

Verified Contracts
==================

By default all the deployable contracts in the given sources are analyzed separately as
the one that will be deployed. This means that if a contract has many direct
and indirect inheritance parents, all of them will be analyzed on their own,
even though only the most derived will be accessed directly on the blockchain.
This causes an unnecessary burden on the SMTChecker and the solver.  To aid
cases like this, users can specify which contracts should be analyzed as the
deployed one. The parent contracts are of course still analyzed, but only in
the context of the most derived contract, reducing the complexity of the
encoding and generated queries. Note that abstract contracts are by default
not analyzed as the most derived by the SMTChecker.

> The chosen contracts can be given via a no-space-comma-separated list of

Reviewer 2: I think the original was hard to read at first, not sure this is better:
```suggestion
The chosen contracts can be given via a comma-separated list (whitespace is not allowed) of
```

<source>:<contract> pairs in the CLI:
``--model-checker-contracts "<source1.sol:contract1>,<source2.sol:contract2>,<source2.sol:contract3>"``,

> and via the object ``settings.modelChecker.contracts`` in the JSON input,

Reviewer 2: "in the JSON input" could link to the standard json section.

option ``--model-checker-engine {all,bmc,chc,none}`` or the JSON option
``settings.modelChecker.engine={all,bmc,chc,none}``.

> 

Reviewer: You intentionally added this empty line?

Author: I think I wrote some stuff there, removed it, and the blank line stayed, Will remove it.
